interfaces/apparmor/apparmor.go

```diff
@@ -117,10 +118,21 @@ func loadProfiles(fnames []string, cacheDir string, flags aaParserFlags) error {
 	if !osutil.GetenvBool("SNAPD_DEBUG") {
 		args = append(args, "--quiet")
 	}
-	args = append(args, fnames...)
 
-	output, err := exec.Command("apparmor_parser", args...).CombinedOutput()
-	if err != nil {
+	parser, requiredArgs, internal, err := apparmor_sandbox.FindAppArmorParser()
+	if err != nil || !internal {
+		// if we couldn't find the parser with apparmor_sandbox
+		// then fall-back to trying to find one in the current PATH
+		// - same for if we are not using the internal
+		// apparmor_parser so that we can support a mocked parser
+		// during tests
+		parser = "apparmor_parser"
```

Comment:
should we emit a `log.Noticef("Using apparmor_parser from $PATH")` or similar, if this can help debugging?

Reply:
I don't think we need to - as part of the apparmor vendoring work I added a new sandbox-feature `parser:snapd-internal` which is then present in the `snap debug sandbox-features` output when using the vendored apparmor - so we can check for this when debugging instead.